src/openssl/lib_crypto.cr

```diff
@@ -1,7 +1,16 @@
 {% begin %}
   lib LibCrypto
-    OPENSSL_110 = {{ `command -v pkg-config > /dev/null && pkg-config --atleast-version=1.1.0 libcrypto || printf %s false`.stringify != "false" }}
-    OPENSSL_102 = {{ `command -v pkg-config > /dev/null && pkg-config --atleast-version=1.0.2 libcrypto || printf %s false`.stringify != "false" }}
+    # An extra zero is appended to the output of LIBRESSL_VERSION to make it 0 when LibreSSL does not exist on the system.
+    # Any comparisons to it should be affixed with an extra zero as well e.g. `(LIBRESSL_VERSION_NUMBER >= 0x2050500F0)`.
+    LIBRESSL_VERSION = {{ `echo "#include <openssl/opensslv.h>\nLIBRESSL_VERSION_NUMBER" | cc -E -`.chomp.split('\n').last.split('L').first.id + "0" }}
```

Comment:
This should be using `CC` from environment and only then fall back to `cc`

Reply:
```crystal
ENV["CC"]? || "cc"
```

Reply:
this is in a shell, just use`${CC:-cc}`.

Reply:
That might need to be revised once this work on non-bash/sh platforms like Windows. Better use now ENV than hunt down that later.

Reply:
It needs to be revised anyway because it uses `cc`... And because it uses openssl. There's a million ways that this snippet is platform specific and this is the absolute least of them.